cvmfs/xattr.cc

```diff
@@ -50,15 +51,27 @@ XattrList *XattrList::CreateFromFile(const std::string &path) {
 
   // Retrieve extended attribute values
   XattrList *result = new XattrList();
-  char value[256];
+  char value_smallbuf[255];
   for (unsigned i = 0; i < keys.size(); ++i) {
     if (keys[i].empty())
       continue;
-    ssize_t sz_value =
-      platform_lgetxattr(path.c_str(), keys[i].c_str(), value, 256);
-    if (sz_value < 0)
-      continue;
-    result->Set(keys[i], string(value, sz_value));
+
+    char *buffer = value_smallbuf;
+    size_t sz_buffer = 255;
+    ssize_t sz_value = platform_lgetxattr(path.c_str(), keys[i].c_str(), buffer,
+                                          sz_buffer);
+    while ((sz_value < 0) && (errno == ERANGE)) {
```

Comment:
Instead of checking repeatedly for ERANGE, I think it should be possible to call lgetxattr with sz_buffer=0, which should then return the size needed for the buffer:

>        If size is specified as zero, these calls return the  current  size  of
>        the  named extended attribute (and leave value unchanged).  This can be
>        used to determine the size of the buffer that should be supplied  in  a
>        subsequent  call.   (But, bear in mind that there is a possibility that
>        the attribute value may change between the two calls,  so  that  it  is
>        still necessary to check the return status from the second call.)

 I'll try this out.